# RNA-Seq
## Methods: 
RNA sequencing data from biological samples ('AD', 'P0', 'P4', 'P7') with replicates ('rep1', 'rep2') was processed using a Snakemake pipeline. The GRCm39 reference genome and GTF annotation file were obtained from the Gencode database (release M34). Quality control analysis of raw sequencing data was conducted using FastQC(v0.12.1). Alignment of reads to the reference genome was performed using STAR aligner(2.7.11b). Gene expression quantification at the exon level was achieved using Verse (v1.0.5). MultiQC was employed to aggregate FastQC results into a single report. Custom Python scripts were utilized for parsing gene annotation, generating a counts matrix from exon-level expression data, and filtering the counts matrix. Further analysis, including differential expression and gene set enrichment analysis, was conducted in the "differential_expression.Rmd" file.
