# Week 4: RNAseq {-}

**Visualizing genes that change over time**

There are **many** ways to analyze time series data including the use of
likelihood ratio tests or spline regression; however, since we have replicates
at each time point, we can also treat these timepoints as factors and explicitly
test for genes that differ between them. Approaches like regression splines
require careful interpretation of curves and choice of parameters. In our
current analysis, we have simply found genes that are differentially expressed
between undifferentiated cells and fully differentiated myocytes. We are going
to utilize two techniques to perform a basic analysis of this data: 1.
Hierarchical clustering + gene enrichment, and 2. Gene Set Enrichment Analysis
(GSEA).

In our analysis, we selected for genes that were differentially expressed
between p0 and AD. Now, we would also like to see how the expression for these
genes changes across time. To do this, we are going to subset our data to only
the DE genes detected between p0 and AD and extract their normalized counts from
each sample. After this, we will mean-center our counts, perform hierarchical
clustering to group them into (hopefully) meaningful clusters and plot their
expression patterns over time.

You may perform all of the following steps in your Rmarkdown notebook. 

  1. Read the DESeq2 vignette, and determine how to generate a Variance Stabilized
    Transformed matrix of your counts
  2. Write this VST counts to a new file or save it to an appropriate variable
  3. You have previously identified genes that are significantly changed between
    p0 and AD. Extract out the genes that are significant at FDR < .001 and subset
    the VST counts to only include the counts for these genes
  4. Perform mean centering but do not scale your data (i.e. subtract the mean 
    value of the row from each value in the row, but do not scale all standard 
    deviations to 1).
  5. Perform hierarchical clustering using the hclust function in R
  6. Use the cutree function in R to cut the dendrogram into distinct clusters.
    - Experiment with different values and find a number that seems appropriate 
      and produces "interesting" clusters
  7. Using the results from cutree, generate separate parallel coordinate plots 
    for each cluster to visualize their expression over time. Make sure you address
    the following questions in your report:
      - What patterns do you see in each cluster?
      - Extract out the genes associated with each of your major clusters, perform
        a quick enrichment analysis using DAVID, ENRICHR, etc. What are some of 
        biological terms enriched in these clusters? How does their expression 
        change over time and rationalize how those changes make sense in the 
        context of the experiment.


For our final analysis, we are going to perform GSEA directly on the results
from the comparison analyzing the p0 and AD timepoints.

  1. Navigate to the official MsigDB website and download the M2 Canonica Pathways
     gene set
     - Upload this .gmt file to your results/ directory on the SCC

As mentioned in class and in BF591, one of the distinct advantages of GSEA is
that it requires no arbitrary statistical threshold and uses the data generated
from all genes in your experiment. You may perform these next steps however you'd
like but we recommend you do so in your existing Rmarkdown notebook.

  1. Use the full dataframe of all results from the p0 vs AD comparison and sort
  your results by log2FoldChange descending
     - Output the values of the log2FoldChange column as a named vector
         (names being the gene symbols)

Read the documentation on the fgsea package in bioconductor.

  1. Using the named vector you generated previously and the .gmt gene set file 
  you downloaded from MSigDB, run fgsea using default parameters
    - fgsea provides a built-in function to load in .gmt files in an appropriate format

  2. Filter your results to only retain gene sets that are significant at a FDR 
  of < .15
  - Using just the significant genesets, generate plots or tables that display 
    the top ten results sorted by both ascending and descending NES. 
    (i.e. one table with the top ten gene sets with the highest NES (most positive),
    and one table with the top ten gene sets that have the lowest NES (most negative)


At the end of the final week, you should have accomplished the following:

- Generated a counts matrix containing variance stabilized transformed counts
- Generated a counts matrix containing VST counts for the DE genes when comparing
  p0 vs AD
- Perform mean centering on these subset of VST counts
- Perform hierarchical clustering on the mean-centered counts
- Used cutree to “cut” our dendrogram into an appropriat enumber of clusters
- Generate parallel coordinate plots for each of the 4 clusters displaying the 
  normalized counts over time for our significant genes
- Performed basic gene enrichment using DAVID/EnrichR on the genes from the 
  "interesting" clusters
- Perform GSEA on the log2FC ranked list of all genes using the fgsea package in 
  R
  - Download the mouse C2 canonical pathways geneset from MSigDB
  - Generate a named vector of log2FoldChanges and gene symbols from all the 
    results returned by DESeq2
  - Filter the fgsea results to only include gene sets with a FDR < .15
  - A plot / table of the top ten gene sets ranked by NES descending
  - A plot / table of the top ten gene sets ranked by NES ascending

