---
title: "Week 3 Notebook"
output: html_document
---


Do your best to write in the style of a scientific publication.
We will spend some time in class discussing some general conventions
and guidelines for scientific writing. 

# Write your methods for this week in the following space:
Quantification
The quantification of RNA-seq data involved the utilization of Aligned files generated by STAR as the input data for each sample. 
The VERSE tool was employed for this task, requiring BAM files and the corresponding GTF file associated with the reference genome. 
Alignments were derived from paired-end reads, and the resulting output files were systematically organized into the results/ directory.

Concatenation of Counts Data
The counts data generated by VERSE were subsequently concatenated into a comprehensive counts matrix using the concat_df.py Python script. 
This script leverages the pandas library, enabling the merging of individual counts columns into a matrix. 
Each column in the resulting matrix represents a sample, and each row corresponds to a unique gene ID.

Mapping Gene IDs to Gene Symbols
The parse_gtf.py script was used to parse the GTF file and extracting a mapping of gene IDs to gene symbols. 
This mapping establishes a meaningful association between numerical gene identifiers and their corresponding human-readable gene names.

Filtering the Counts Matrix
To improve the reliability of downstream analyses, the counts matrix underwent a filtering process using the filter_cts_mat.py script. 
This script selectively removed genes that exhibited no expression across all samples, ensuring the creation of a consistent dataset for subsequent statistical analyses.

# Write your results for this week in the following space:

The analysis resulted in several key outputs:

Exon-level Expression Matrices: Individual exon-level expression matrices were generated for each sample using VERSE (results/{sample}.exon.txt).

Full Expression Matrix: The exon-level expression matrices were concatenated into a comprehensive matrix (results/matrix.csv). 
This matrix provides a consolidated view of gene expression across all samples.

Gene Mapping Information: The parse_gtf.py script was employed to extract and map gene IDs to gene symbols, generating a CSV file (results/genemap.csv).

Filtered Count Matrix: The counts matrix underwent a filtering process using the filter_cts_mat.py script, resulting in a filtered count matrix (results/filtered_counts_matrix.csv). 

Since subsampled files were utilized, the counts values for every sample were uniformly 0 and consequently, the filtered counts matrix also yielded 0 rows.

```{r, if you need any code blocks, you can instantiate them like this}


```


# Conceptual Questions

